src/btree/bt_sync.c

```diff
@@ -108,11 +108,26 @@ static bool
 __sync_ref_is_obsolete(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_ADDR *addr;
+    WT_CELL_UNPACK vpack;
 
-    addr = ref->addr;
+    /* Ignore root pages as they can never be deleted. */
+    if (__wt_ref_is_root(ref))
+        return (false);
+
+    /* Guard against marking an in-memory page to be deleted. */
+    if (ref->state != WT_REF_DISK)
+        return (false);
+
+    /* Ignore internal pages, these are taken care of during reconciliation. */
+    if (!__wt_ref_is_leaf(session, ref))
+        return (false);
 
-    return (
-      addr != NULL && __wt_txn_visible_all(session, addr->newest_stop_txn, addr->newest_stop_ts));
+    WT_ORDERED_READ(addr, ref->addr);
+    if (!__wt_off_page(ref->home, addr)) {
+        __wt_cell_unpack(session, ref->home, (WT_CELL *)addr, &vpack);
+        return (__wt_txn_visible_all(session, vpack.newest_stop_txn, vpack.newest_stop_ts));
+    } else
+        return (__wt_txn_visible_all(session, addr->newest_stop_txn, addr->newest_stop_ts));
```

Comment:
There's no need for an `else` after a `return`.

Reply:
fixed.